pkg/filestats/filestats.go

```diff
@@ -0,0 +1,74 @@
+package filestats
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/wakatime/wakatime-cli/pkg/heartbeat"
+
+	jww "github.com/spf13/jwalterweatherman"
+)
+
+// Max file size supporting line number count stats. Files larger than this in
+// bytes will not have a line count stat for performance. Default is 2MB (2*1024*1014).
+const maxFileSizeSupported = 2097152
+
+// WithDetection initializes and returns a heartbeat handle option, which
+// can be used in a heartbeat processing pipeline to detect filestats. At the
+// moment only the total number of lines in a file is detected.
+func WithDetection() heartbeat.HandleOption {
+	return func(next heartbeat.Handle) heartbeat.Handle {
+		return func(hh []heartbeat.Heartbeat) ([]heartbeat.Result, error) {
+			for n, h := range hh {
+				if h.EntityType == heartbeat.FileType {
+					fileInfo, err := os.Stat(h.Entity)
+					if err != nil {
+						jww.ERROR.Printf("failed to retrieve file stats of file %q: %s", h.Entity, err)
+						continue
+					}
+
+					if fileInfo.Size() > maxFileSizeSupported {
```

Comment:
Isn't there a way to test this validation? I believe uploading a file with more than 2mb to test it will increase the download time for this repository. What do you think?

Reply:
@gandarez I added a test, which will create a temporary file, which exceeds the size.

Reply:
good!